csharp/src/Microsoft.ML.OnnxRuntime/DisposableNamedOnnxValue.cs

```diff
@@ -60,13 +61,48 @@ public void Dispose()
 
     public class DisposableNamedOnnxValue : NamedOnnxValue, IDisposable
     {
-        protected IDisposable _nativeMemoryManager;
-        protected DisposableNamedOnnxValue(string name, Object value, IDisposable nativeMemoryManager)
+        private NativeMemoryHandler _nativeMemoryManager;
+        private DisposableNamedOnnxValue(string name, Object value, NativeMemoryHandler nativeMemoryManager)
             : base(name, value)
         {
             _nativeMemoryManager = nativeMemoryManager;
         }
 
+        /// <summary>
+        /// Overrides the base class method. Since the instance already has access to the 
+        /// underlying OrtValue handle (if this instance hasn't been disposed), it just assigns
+        /// that to the output onnxValue. With respect to pinnedMemoryHandle, it has no operation
+        /// to do, as this class doesn't maintain a managed buffer. It doesn't have to maintain it
+        /// as it already is associated with the object of interest (native OrtValue)
+        /// </summary>
+        /// <param name="onnxValue"></param>
+        /// <param name="pinnedMemoryHandle"></param>
+        /// <param name="disposeOnnxValueAfterUse"></param>
+        internal override void ToNativeOnnxValue(out IntPtr onnxValue, 
+                                                 out MemoryHandle pinnedMemoryHandle,
+                                                 out bool disposeOnnxValueAfterUse)
+        {
+            // Make sure that this instance hasn't been disposed yet
+            if (disposedValue)
+            {
+                throw new Exception("This instance of DisposableNamedOnnxValue has already been disposed");
```

Comment:
>This instance of DisposableNamedOnnxValue has already been disposed [](start = 37, length = 67)

Should be a bug?

Reply:
bug in the user code ? Probably. But they get a user friendly error message this way...